pkg/clientgen/testdata/tsapp/expected_typescript.ts

```diff
@@ -43,6 +45,19 @@ export default class Client {
         const base = new BaseClient(target, options ?? {})
         this.svc = new svc.ServiceClient(base)
     }
+
+    /**
+     * Creates a new Encore client with the given auth handler.
+     *
+     * @param authHandler Authentication data to be used for each request. Either a static  
+     *                    object or a function which returns a new object for each request.
+     **/
+    public withAuth(authHandler: svc.AuthParams | AuthDataGenerator): Client {
```

Comment:
Should the argument also support `| undefined` to unset the auth generator?

We should probably not call it `authHandler` since that is another Encore concept